pkg/api/pod/warnings_test.go

```diff
@@ -1058,6 +1058,84 @@ func TestWarnings(t *testing.T) {
 				`spec.containers[1].ports[1]: duplicate port definition with spec.containers[1].ports[0]`,
 			},
 		},
+		{
+			name: "create duplicate container ports name in two containers",
+			template: &api.PodTemplateSpec{Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name: "foo1",
+						Ports: []api.ContainerPort{
+							{ContainerPort: 80, HostPort: 80, Protocol: api.ProtocolUDP, Name: "test"},
+							{ContainerPort: 180, HostPort: 180, Protocol: api.ProtocolUDP},
+							{ContainerPort: 8080, HostPort: 8080, Protocol: api.ProtocolUDP, Name: "test"},
+						},
+					},
+					{
+						Name: "foo",
+						Ports: []api.ContainerPort{
+							{ContainerPort: 8090, HostPort: 8090, Protocol: api.ProtocolUDP, Name: "test"},
+							{ContainerPort: 8091, HostPort: 8091, Protocol: api.ProtocolUDP, Name: "test"},
+						},
+					}},
+			}},
+			expected: []string{
+				`spec.containers[0].ports[2]: duplicate port name definition with spec.containers[0].ports[0]`,
+				`spec.containers[1].ports[0]: duplicate port name definition with spec.containers[0].ports[0]`,
+				`spec.containers[1].ports[0]: duplicate port name definition with spec.containers[0].ports[2]`,
```

Comment:
we don't want multiple warnings about the same field... just remember the first time we see each port name defined and have a single warning for duplicating that field